apps/explorer/src/modules/bridge/hooks/useBridgeProviderBuyTokens.ts

```diff
@@ -0,0 +1,29 @@
+import { SWR_NO_REFRESH_OPTIONS } from '@cowprotocol/common-const'
+import type { CrossChainOrder } from '@cowprotocol/cow-sdk'
+import type { TokenInfo } from '@uniswap/token-lists'
+
+import useSWR, { SWRResponse } from 'swr'
+
+export function useBridgeProviderBuyTokens(
+  provider: CrossChainOrder['provider'] | undefined,
+  chainId: number,
+): SWRResponse<Record<string, TokenInfo> | undefined> {
+  return useSWR(
+    [provider, chainId, provider?.info.dappId],
+    async ([provider, chainId]) => {
```

Comment:
You mentioned in another PR that having the provider here is bad. Should we still keep it?